src/Raven.Server/Documents/Sharding/Handlers/Processors/Documents/ShardedDocumentHandlerProcessorForGet.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Primitives;
+using Raven.Client.Documents.Operations.TimeSeries;
+using Raven.Server.Documents.Handlers.Processors.Documents;
+using Raven.Server.Documents.Queries.Revisions;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.Json;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Documents;
+
+internal class ShardedDocumentHandlerProcessorForGet : AbstractDocumentHandlerProcessorForGet<ShardedDocumentHandler, TransactionOperationContext, BlittableJsonReaderObject>
+{
+    private readonly OperationCancelToken _operationCancelToken;
+
+    public ShardedDocumentHandlerProcessorForGet(HttpMethod method, [NotNull] ShardedDocumentHandler requestHandler) : base(method, requestHandler)
+    {
+        _operationCancelToken = requestHandler.CreateOperationToken();
+    }
+
+    protected override bool SupportsShowingRequestInTrafficWatch => false;
+
+    protected override bool SupportsHandlingOfMissingIncludes => true;
+
+    protected override CancellationToken CancellationToken => _operationCancelToken.Token;
+
+    protected override void Initialize(TransactionOperationContext context)
+    {
+    }
+
+    protected override async ValueTask<DocumentsByIdResult<BlittableJsonReaderObject>> GetDocumentsByIdImplAsync(TransactionOperationContext context, StringValues ids,
+        StringValues includePaths, RevisionIncludeField revisions, StringValues counters, HashSet<AbstractTimeSeriesRange> timeSeries, StringValues compareExchangeValues, bool metadataOnly, string etag)
+    {
+        DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal, "make sure we maintain the order of returned results");
+
+        DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Arek, DevelopmentHelper.Severity.Normal, "Includes of revisions / counters / time series / compare exchanges");
+
+        if (counters.Count > 0)
+            throw new NotSupportedException("Include of counters is not supported");
+
+        if (revisions != null)
+            throw new NotSupportedException("Include of revisions is not supported");
+
+        if (timeSeries != null)
+            throw new NotSupportedException("Include of time series is not supported");
+
+        if (compareExchangeValues.Count > 0)
+            throw new NotSupportedException("Include of compare exchange is not supported");
```

Comment:
Let's throw `NotSupportedInShardingException` for now

Reply:
Ok. Changed to NotSupportedInShardingException